### PR TITLE
Add in missing overrides.

### PR DESCRIPTION
**Pull request by chrisdembia**
This gets rid of tons of clang warnings.

I'd like to merge this soon to avoid merge conflicts.

### CODE REVIEW

File: OpenSim/Actuators/ContDerivMuscle_Deprecated.h [hunk 1]
```diff
@@ -131,23 +131,23 @@ OpenSim_DECLARE_CONCRETE_OBJECT(ContDerivMuscle_Deprecated,
...
+    virtual void setActivationDeriv(const SimTK::State& s, double activationDeriv) const override { setStateVariableDeriv(s, STATE_ACTIVATION_NAME, activationDeriv); }
+    virtual double getFiberLength(const SimTK::State& s) const override { return getStateVariableValue(s, STATE_FIBER_LENGTH_NAME); }
+    virtual void setFiberLength(SimTK::State& s, double fiberLength) const override { setStateVariableValue(s, STATE_FIBER_LENGTH_NAME, fiberLength); }
+    virtual double getFiberLengthDeriv(const SimTK::State& s) const override { return getStateVariableDeriv(s, STATE_FIBER_LENGTH_NAME); }
+    virtual void setFiberLengthDeriv(const SimTK::State& s, double fiberLengthDeriv) const override { setStateVariableDeriv(s, STATE_FIBER_LENGTH_NAME, fiberLengthDeriv); }
+    virtual void setTendonForce(const SimTK::State& s, double aForce) const override;
+    virtual double getTendonForce( const SimTK::State& s) const override;
     virtual void setActiveForce(const SimTK::State& s, double aForce) const;
     virtual double getActiveForce( const SimTK::State& s) const;
```
Comment (sherm1) [inline OpenSim/Actuators/ContDerivMuscle_Deprecated.h]: Suspicious that these are not overrides but set/getPassiveForce() are.

File: OpenSim/Actuators/ContDerivMuscle_Deprecated.h [hunk 2]
```diff
@@ -158,13 +158,13 @@ OpenSim_DECLARE_CONCRETE_OBJECT(ContDerivMuscle_Deprecated,
...
-    virtual void computeEquilibrium(SimTK::State& s ) const;
-    virtual double computeActuation(const SimTK::State& s) const;
+    virtual void computeEquilibrium(SimTK::State& s ) const override;
+    virtual double computeActuation(const SimTK::State& s) const override;
     double calcTendonForce(const SimTK::State& s, double aNormTendonLength) const;
-    double calcPassiveForce(const SimTK::State& s, double aNormFiberLength) const;
-    double calcActiveForce(const SimTK::State& s, double aNormFiberLength) const;
+    double calcPassiveForce(const SimTK::State& s, double aNormFiberLength) const override;
+    double calcActiveForce(const SimTK::State& s, double aNormFiberLength) const override;
     double calcFiberVelocity(const SimTK::State& s, double aActivation, double aActiveForce, double aVelocityDependentForce) const;
```
Comment (sherm1) [inline OpenSim/Actuators/ContDerivMuscle_Deprecated.h]: calcTendonForce/calcFiberVelocity not overrides?

File: OpenSim/Common/Storage.h [hunk 3]
```diff
@@ -152,19 +152,19 @@ OpenSim_DECLARE_CONCRETE_OBJECT(Storage, StorageInterface);
     // GET AND SET
     //--------------------------------------------------------------------------
     // SIZE
-    virtual int getSize() const { return(_storage.getSize()); }
+    virtual int getSize() const override { return(_storage.getSize()); }
```
Comment (sherm1) [inline OpenSim/Common/Storage.h]: Harmless, but this shouldn't have both `virtual` and `override` (several more below).

**Comment by sherm1** [comment]
> This gets rid of tons of clang warnings.

What warning does it get rid of? I didn't know there was a warning for leaving off override.

**Comment by sherm1** [comment]
Done reviewing. There are lots of cases now that say `virtual blah() override;` which doesn't adhere to our coding conventions. If there is a reasonable way to get rid of those now-redundant `virtual` keywords I think it would be good to do so now because we are unlikely to do it later. But if it is too daunting, I'm OK with leaving them since I think it is still an improvement to have `override` added.

**Comment by chrisdembia** [comment]
> What warning does it get rid of? I didn't know there was a warning for leaving off override.

This PR is analogous to the one I did for simbody, which was to get rid of the same warning. The warning is `Winconsistent-missing-override`.

**Comment by sherm1** [comment]
> The warning is Winconsistent-missing-override.

Oh yeah -- thanks!

**Comment by chrisdembia** [comment]
Thanks for reviewing. I'll take a shot at removing the virtuals and addressing your other comments.

**Comment by chrisdembia** [comment]
Oh gosh... this PR adds "override" to a bunch of constructProperty methods, which is undesired. We'll have to be careful with this PR.

**Comment by sherm1** [comment]
> Oh gosh... this PR adds "override" to a bunch of constructProperty methods, which is undesired. We'll have to be careful with this PR.

Why is that undesired? Doesn't that mean the methods were really overriding something?

**Comment by chrisdembia** [comment]
> Why is that undesired? Doesn't that mean the methods were really overriding something?

`constructProperties()` is currently virtual, but originally it was not; we are hoping to go back to that.

**Pull request closed without merging**